crates/http/src/http.rs

```diff
@@ -60,33 +62,26 @@ pub trait HttpClient: Send + Sync {
         }
     }
 
-    fn proxy(&self) -> Option<&Uri>;
+    fn proxy(&self) -> Proxy;
 }
 
 /// An [`HttpClient`] that may have a proxy.
-#[derive(Deref)]
 pub struct HttpClientWithProxy {
-    #[deref]
     client: Arc<dyn HttpClient>,
```

Comment:
The client no longer has a default proxy settings now, should we keep the `#[Deref]`?